### PR TITLE
[devicelab] introduce new old gallery.

### LINKED ISSUE #143482: [framework] reduce dependencies and move archived flutter/gallery into framework repo for benchmarking.
We'd like to preserve these benchmark cases, even if the application itself is no longer updated. To do this, we'll need to reduce the set of dependencies and possibly clean up some of the assets.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/143482

This brings in the gallery more or less as is:

* Removed localizations
* Ensure tests still run (locally verified, will switch CI later).
* Removed deferred components
* Fixup pubspec

### CODE REVIEW

File: dev/integration_tests/new_gallery/analysis_options.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+include: package:flutter_lints/flutter.yaml
```
Comment (goderbauer) [inline dev/integration_tests/new_gallery/analysis_options.yaml:1]: Fly-by comment: Can we delete this file to apply the normal flutter lint configuration to the gallery and enforce a unified coding style across this repo?

Comment (jonahwilliams) [reply dev/integration_tests/new_gallery/analysis_options.yaml:1]: `always_specify_types` is a big lift here.

Comment (goderbauer) [reply dev/integration_tests/new_gallery/analysis_options.yaml:1]: According to https://dart.dev/tools/linter-rules/always_specify_types there's an auto-fix available for that lint. Presumably, running that over the code base here should fix most (if not all) cases automatically?

Comment (jonahwilliams) [reply dev/integration_tests/new_gallery/analysis_options.yaml]: dart fix got some things, but as of now:

```
jonahwilliams-macbookpro3:new_gallery jonahwilliams$ dart fix --dry-run
Computing fixes in new_gallery (dry run)... 3.4s
Nothing to fix!
```

Comment (jonahwilliams) [reply dev/integration_tests/new_gallery/analysis_options.yaml]: I think that just means I can click on a call site in the IDE to have it insert the type?

Comment (jonahwilliams) [reply dev/integration_tests/new_gallery/analysis_options.yaml]: actually dart fix broke all the null checks 😢

**Comment by jonahwilliams** [comment]
I'll probably split this in two. one that rolls in the gallery, and a second to update the benchmarks.

File: dev/integration_tests/new_gallery/analysis_options.yaml (hunk 1 above)
Comment (jonahwilliams) [reply dev/integration_tests/new_gallery/analysis_options.yaml]: lints fixed

**Review by zanderso — APPROVED** [review]
lgtm. @goderbauer should also approve when ready.

File: dev/integration_tests/new_gallery/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+  web: 0.4.2 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
+  win32: 5.2.0 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
+  xdg_directories: 1.0.4 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
+
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+  flutter_driver:
+    sdk: flutter
+  flutter_lints: 3.0.1
```
Comment (goderbauer) [inline dev/integration_tests/new_gallery/pubspec.yaml:71]: nit: this one can probably be removed now?

**Review by goderbauer — APPROVED** [review]
LGTM

Thank you for also doing the lints clean-up!

File: dev/integration_tests/new_gallery/pubspec.yaml (hunk 2 above)
Comment (jonahwilliams) [reply dev/integration_tests/new_gallery/pubspec.yaml]: Done

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**